public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx

```diff
@@ -0,0 +1,337 @@
+      savedAugmentObjects.filter(
+        (savedObj) => get(savedObj, 'visId', '') === embeddable.vis.id
+      );
+
+    // Map all detector IDs for all the found augmented vis objects
+    const savedAugmentDetectorsSet = new Set(
+      savedAugmentForThisVisualization.map((savedObject) =>
+        get(savedObject, 'pluginResourceId', '')
+      )
+    );
+
+    // filter out any detectors that aren't on the set of detectors IDs from the augmented vis objects.
+    const detectorsToDisplay = detectors.filter((detector) =>
+      savedAugmentDetectorsSet.has(detector.id)
+    );
+    return detectorsToDisplay;
+  };
+
+  const onUnlinkDetector = async () => {
+    setIsLoadingFinalDetectors(true);
+    await savedObjectLoader.findAll().then(async (resp: any) => {
+      if (resp != undefined) {
+        const savedAugmentObjects: ISavedAugmentVis[] = get(resp, 'hits', []);
+        // gets all the saved object for this visualization
+        const savedAugmentForThisVisualization: ISavedAugmentVis[] =
+          savedAugmentObjects.filter(
+            (savedObj) => get(savedObj, 'visId', '') === embeddable.vis.id
+          );
+
+        // find saved Augment object matching detector we want to unlink
+        // There should only be one detector and vis pairing
+        const savedAugmentToUnlink = savedAugmentForThisVisualization.find(
+          (savedObject) =>
+            get(savedObject, 'pluginResourceId', '') === detectorToUnlink.id
+        );
+        const savedObjectToUnlinkId = get(savedAugmentToUnlink, 'id', '');
+        await savedObjectLoader
+          .delete(savedObjectToUnlinkId)
+          .then(async (resp: any) => {
+            core.notifications.toasts.addSuccess({
+              title: `Association removed between the ${detectorToUnlink.name}
+              and the ${embeddableTitle} visualization`,
+              text: "The detector's anomalies do not appear on the visualization. Refresh your dashboard to update the visualization",
+            });
+          })
+          .catch((error) => {
+            core.notifications.toasts.addDanger(
+              prettifyErrorMessage(
+                `Error unlinking selected detector: ${error}`
+              )
+            );
+          })
+          .finally(() => {
+            getDetectors();
+          });
+      }
+    });
+  };
+
+  const handleHideModal = () => {
+    setConfirmModalState({
+      ...confirmModalState,
+      isOpen: false,
+    });
+  };
+
+  const handleConfirmModal = () => {
+    setConfirmModalState({
+      ...confirmModalState,
+      isRequestingToClose: true,
+    });
+  };
+
+  const getDetectors = async () => {
+    dispatch(getDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS));
+  };
+
+  // TODO: this part is incomplete because it is pending on a different PR that will have all the associate existing changes
+  const openAssociateDetectorFlyout = async () => {
+    console.log('inside create anomaly detector');
+  };
```

Comment:
Is this pending once create detector flyout is merged first?

Reply:
and this: https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/484